src/Config/ExtensionPluginInterface.php

```diff
@@ -17,9 +17,10 @@ interface ExtensionPluginInterface
     /**
      * Allows a plugin to override extension configuration.
      *
-     * @param string $extensionName
+     * @param string                    $extensionName
+     * @param list<array<string,mixed>> $extensionConfigs
      *
-     * @return array<string,mixed>
+     * @return list<array<string,mixed>>
```

Comment:
```suggestion
     * @return list<string,mixed>
```

Otherwise it is one array level too much, isn‘t it?

Reply:
`list<string,mixed>` is invalid, because `list` can only have one type inside `<…>` (no type for the key as this is always `int` for lists)

I think `list<array<string,mixed>>` is correct because it seems to describe a structure like this:

```php
[
    [
        'string1' => …mixed…,
        'string2' => …mixed…,
    ],
    [
        'string3' => …mixed…,
    ],
]
```